azure/services/virtualmachines/client.go

```diff
@@ -109,14 +109,21 @@ func (ac *AzureClient) CreateOrUpdateAsync(ctx context.Context, spec azure.Resou
 // request to Azure and if accepted without error, the func will return a Poller which can be used to track the ongoing
 // progress of the operation.
 func (ac *AzureClient) DeleteAsync(ctx context.Context, spec azure.ResourceSpecGetter, resumeToken string) (poller *runtime.Poller[armcompute.VirtualMachinesClientDeleteResponse], err error) {
-	ctx, _, done := tele.StartSpanWithLogger(ctx, "virtualmachines.AzureClient.Delete")
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "virtualmachines.AzureClient.Delete")
 	defer done()
 
 	forceDelete := ptr.To(true)
 	opts := &armcompute.VirtualMachinesClientBeginDeleteOptions{ResumeToken: resumeToken, ForceDeletion: forceDelete}
 	poller, err = ac.virtualmachines.BeginDelete(ctx, spec.ResourceGroupName(), spec.ResourceName(), opts)
 	if err != nil {
-		return nil, err
+		if azure.BadRequest(err) {
+			log.Info("Failed to Begin VM Delete with Force Deletion, retrying without the force flag")
+			opts.ForceDeletion = ptr.To(false)
+			poller, err = ac.virtualmachines.BeginDelete(ctx, spec.ResourceGroupName(), spec.ResourceName(), opts)
+		}
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This looks like a good improvement, but is it related to Azure Stack support? What problem is it trying to solve?

Reply:
Yes, azure stack throws a 400 error that says the force flag is not supported... Should I add a comment?

Reply:
oops I think I forgot to add the comment. This is specified in the commit message, but I will add a code comment in my next pass.